lib/api/core/models/repositories/tokenRepository.js

```diff
@@ -76,14 +76,12 @@ module.exports = function (kuzzle) {
 
     if (!user || user._id === null) {
       error = new InternalError('Unknown User : cannot generate token');
-      deferred.reject(error);
-      return deferred.promise;
+      return q.reject(error);
```

Comment:
Minor/nitpicking: In that case, the line `deferred = q.defer()` should be put after these checks, to avoid creating a promise whose sole purpose is to be garbage collected.